### PR TITLE
fix: Execute alert-related jobs even if trivyscan is skipped

### PR DESCRIPTION
**Pull request by mmcgehee-ccc**

### CODE REVIEW

File: .github/workflows/golang-security-scan.yml [hunk 1]
```diff
@@ -33,46 +33,41 @@ jobs:
...
           severity: "MEDIUM,HIGH,CRITICAL"
-    outputs:
-      status: ${{ job.status }}
 
   create_issues:
     name: Create issues for failed jobs
-    needs: [govulncheck, trivyscan]
     runs-on: ubuntu-latest
+    needs: [govulncheck, trivyscan]
+    if: always() && (needs.govulncheck.result == 'failure' || needs.trivyscan.result == 'failure')
```
Comment (jwatson-ccc) [inline .github/workflows/golang-security-scan.yml:68]: More curious than anything, why do we need `always()` here?

Comment (mmcgehee-ccc) [reply .github/workflows/golang-security-scan.yml:68]: Without it the job is skipped if either of the jobs listed for 'needs' are skipped.

**Comment by mmcgehee-ccc** [comment]
@jwatson-ccc 
This one isn't on our board, but it's ready for review when you have time.

**Review by jwatson-ccc — APPROVED** [review]
LGTM

**Pull request merged**